app/hooks/use-mobile.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react";
+
+const MOBILE_BREAKPOINT = 768;
+
+export function useIsMobile() {
+  const [isMobile, setIsMobile] = useState<boolean | undefined>(undefined);
+
+  useEffect(() => {
+    const mql = window.matchMedia(`(max-width: ${MOBILE_BREAKPOINT - 1}px)`);
```

Comment:
I would wrap all of this in `if (window)` so we dont get hydration issues because of SSR